### PR TITLE
What fuzzing could look like

### PR DESCRIPTION
**Pull request by 1-alex98**
Just trying

### CODE REVIEW

File: faf-commons-data/src/main/java/com/faforever/commons/map/PreviewGenerator.java [hunk 1]
```diff
@@ -71,13 +85,16 @@ private static MapData parseMap(Path mapPath) {
       int ddsSize = mapInput.readInt();
       // Skip DDS header
       mapInput.skipBytes(128);
-
+      if(ddsSize<128 || ddsSize > 1_000_000){
```
Comment (1-alex98) [inline faf-commons-data/src/main/java/com/faforever/commons/map/PreviewGenerator.java:88]: Otherwise produces OOME .. should be fixed

**Comment by 1-alex98** [comment]
https://github.com/FAForever/faf-java-commons/pull/93/files#r917404405

**Comment by 1-alex98** [comment]
The images added here do actually belong in this repo instead of the java-api

File: faf-commons-data/build.gradle [hunk 2]
```diff
@@ -5,7 +5,18 @@ dependencies {
   implementation("com.google.guava:guava:${guavaVersion}")
   implementation("com.fasterxml.jackson.core:jackson-databind:${jacksonVersion}")
   implementation("com.github.luben:zstd-jni:${zstdVersion}")
+  testImplementation 'com.code-intelligence:jazzer-api:0.11.0'
   api("org.apache.commons:commons-compress:${apacheCommonsCompressVersion}")
 
   testImplementation("ch.qos.logback:logback-classic:${logbackVersion}")
 }
+
+task fuzzFatJar(type: Jar) {
```
Comment (1-alex98) [inline faf-commons-data/build.gradle:14]: builds a fat jar with the test classes

File: faf-commons-data/src/test/java/com/faforever/commons/map/PreviewGeneratorFuzzTarget.java [hunk 3]
```diff
@@ -0,0 +1,335 @@
...
+      try {
+        PreviewGenerator.generatePreview(
+          Path.of("."), 256, 256
+        );
+        /*try{
+          ImageIO.write(bufferedImage, "png", Path.of("./"+Math.random()+".png").toFile());
+        }catch (Exception e){
+          throw new Error(e);
+        }*/
+      } catch (PreviewGenerator.IllegalInput ignore) {
```
Comment (adorne) [inline faf-commons-data/src/test/java/com/faforever/commons/map/PreviewGeneratorFuzzTarget.java:331]: Are these commented lines really needed?

Comment (1-alex98) [reply faf-commons-data/src/test/java/com/faforever/commons/map/PreviewGeneratorFuzzTarget.java:331]: It is just for fun to see what the generated images look like

Comment (adorne) [reply faf-commons-data/src/test/java/com/faforever/commons/map/PreviewGeneratorFuzzTarget.java:331]: I think that JavaDoc is a better place for this.

**Comment by Brutus5000** [comment]
closed due to inactivity

**Pull request closed without merging**